app/main.py

```diff
@@ -1,3 +1,16 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Any
+
+
+def cache(func: Callable) -> Callable:
+    result_store = {}
+
+    def wrapper(*args) -> Any:
+        if args not in result_store:
+            print("Calculating new result")
+            result_store[args] = func(*args)
+            return result_store.get(args)
+        else:
+            print("Getting from cache")
+            return result_store.get(args)
```

Comment:
You're using return statement twice. Try to use it just once

Reply:
Fixed, I think))